### DISCUSSION TITLE
Dynamic matrix yaml syntax error

### DISCUSSION THREAD

**Discussion by chingc** [community/community#25475] [Actions]
<p>I’m suddenly getting “You have an error in your yaml syntax” when using a dynamically generated matrix.</p>
<p>Even the example given from the <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#fromjson" rel="noopener nofollow ugc">fromJson docs</a> is saying there’s a syntax error.</p>
<p>It says it’s in job2 where the matrix is defined, but I don’t see a problem.</p>
<pre><code class="lang-auto">name: test
on: push

jobs:
  job1:
    runs-on: ubuntu-latest
    outputs:
      matrix: ${{ steps.set-matrix.outputs.matrix }}
    steps:
    - id: set-matrix
      run: echo "::set-output name=matrix::{\"include\":[{\"project\":\"foo\",\"config\":\"Debug\"},{\"project\":\"bar\",\"config\":\"Release\"}]}"

  job2:
    needs: job1
    runs-on: ubuntu-latest
    strategy:
      matrix: ${{fromJson(needs.job1.outputs.matrix)}}
    steps:
    - run: echo hello
</code></pre>

**Comment by sindrig** [comment]
<p>+1 Been going crazy for the last hour trying to figure out how I was able to break this.</p>

**Comment by Croydon** [comment]
<p>Same here. Our GitHub Actions broke without any change from our side.</p>

**Comment by jvisker** [comment]
<p>We are having problems as well</p>

**Comment by chingc** [comment]
<p><a class="mention" href="/sindrig">@sindrig</a> <a class="mention" href="/croydon">@croydon</a> <a class="mention" href="/jvisker">@jvisker</a> This started working again for me this morning.  I didn’t make any changes on my end.  Would be nice to get some kind of explanation from GitHub, but I feel like this didn’t affect enough users for them to make a statement.</p>

**Comment by sindrig** [comment]
<p>Same for us. I opened a ticket with GitHub support last night and got a reply 4 hours later that they thought this was fixed.</p>
<p>I asked them for an explanation but have not received anything. I’ll update here if I get an answer.</p>

**Comment by cschleiden** [comment, marked as answer]
<p>This should be fixed now, unfortunately there was a very brief time window when the parser reported a syntax error for this.</p>
